src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -62,40 +70,93 @@ public ExternalFilesEntryLinker getLinker() {
         return linker;
     }
 
-    public void importAsNewEntries(List<Path> files) {
-        CompoundEdit ce = new CompoundEdit();
-        for (Path file : files) {
-            List<BibEntry> entriesToAdd;
-            if (FileUtil.getFileExtension(file).filter("pdf"::equals).isPresent()) {
-                List<BibEntry> pdfResult = contentImporter.importPDFContent(file);
-                List<BibEntry> xmpEntriesInFile = contentImporter.importXMPContent(file);
-
-                // First try xmp import, if empty try pdf import, otherwise create empty entry
-                if (!xmpEntriesInFile.isEmpty()) {
-                    if (!pdfResult.isEmpty()) {
-                        // FIXME: Show merge dialog?
-                        entriesToAdd = xmpEntriesInFile;
-                    } else {
-                        entriesToAdd = xmpEntriesInFile;
+    public BackgroundTask<List<ImportFilesResultItemViewModel>> importFilesInBackground(List<Path> files) {
+        return new BackgroundTask<>() {
+
+            @Override
+            protected List<ImportFilesResultItemViewModel> call() {
+                results = new ArrayList<>();
+                counter = 1;
+                CompoundEdit ce = new CompoundEdit();
+                for (int i = 0; i < files.size(); i++) {
+
+                    var file = files.get(i);
+                    entriesToAdd = Collections.emptyList();
+
+                    if (isCanceled()) {
+                        break;
                     }
-                } else {
-                    if (!pdfResult.isEmpty()) {
-                        entriesToAdd = pdfResult;
-                    } else {
-                        entriesToAdd = Collections.singletonList(createEmptyEntryWithLink(file));
+                    DefaultTaskExecutor.runInJavaFXThread(() -> {
+                        updateMessage(Localization.lang("Processing file %0 of %1", counter, files.size()));
+                        updateProgress(counter, files.size() - 1);
+                    });
+
+                    try {
+                        if (FileUtil.getFileExtension(file).filter("pdf"::equals).isPresent()) {
```

Comment:
I think it would make sense to introduce a `FileUtil.isPdfFile` similar to the isbibfile method.